core/vm/interface.go

```diff
@@ -26,58 +26,88 @@ import (
 
 // StateDB is an EVM database for full state querying.
 type StateDB interface {
+	// CreateAccount creates a new account
 	CreateAccount(common.Address)
-
+	// SubBalance subtracts the amount from the address's balance
 	SubBalance(common.Address, *big.Int)
+	// AddBalance adds the amount to the address's balance
 	AddBalance(common.Address, *big.Int)
+	// GetBalance returns the balance for the given address
+	// For non-existent addresses or self-destructed addresses (during the tx execution)
+	// it returns zero
 	GetBalance(common.Address) *big.Int
-
+	// GetNonce returns the nonce for the given address
+	// For non-existent addresses it returns zero
 	GetNonce(common.Address) uint64
+	// SetNonce sets the nonce for the given address
 	SetNonce(common.Address, uint64)
-
+	// GetCodeHash returns the codeHash for the given address
+	// For non-existent addresses or addresses without a code (EOAs) it returns types.EmptyCodeHash
```

Comment:
This is not correct. For non-existent addresses, this returns (common.Hash{})

For existing accounts, but without code, this returns types.EmptyCodeHash